### PR TITLE
For Java 20, JSP adjust option jdkSourceLevel up to level 8 if not already

### LINKED ISSUE #23885: Use mininum jdkSourceLevel of 1.8 for JDK 20+
**Describe the bug**  
In JDK20, javac has removed support for Java 1.7 compiling.  JSP needs an update to the code that checks jdkSourceLevel.  Until this is done errors like the following will happen when jdkSourceLevel="17" (meaning Java 1.7) is specified:

com.ibm.ws.jsp.JspCoreException: JSPG0049E: /xxxxxxxxx.jsp failed to compile :
error: Source option 7 is no longer supported. Use 8 or later.

**Steps to Reproduce**  
See above.

**Expected behavior**  
The jdkSourceLevel specified by the user will be overrided to "18" if a lower level is specified, when Java 20 or higher is the runtime.

### PR DESCRIPTION
**Pull request by epj**
fixes #23885

### CODE REVIEW

**Comment by epj** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp_fat

**Comment by epj** [comment]
The build above had only 2 test errors, both in UI.   I also have a Java 20 build running that I'll post the results of when it is done.

File: dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/JspOptions.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 1997, 2004 IBM Corporation and others.
+ * Copyright (c) 1997, 2004, 2023 IBM Corporation and others.
```
Comment (volosied) [inline dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/JspOptions.java]: Remove '2004,`

**Comment by epj** [comment]
Java 20 build:   https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_XEvIQJZvEe2YL83b0ePHDw&tab=com.ibm.team.build.web.ui.internal.editors.result.overview.OverviewPage

There were no jsp related failures. (Without this change there are two JSP FAT failures.)

**Comment by epj** [comment]
#libbybot

**Review by volosied — APPROVED** [review]
LGTM

**Pull request merged**